shed/translation.py

```diff
@@ -0,0 +1,172 @@
+import uuid
+import time
+
+from streamz.core import Stream
+from databroker._core import ALL
+import networkx as nx
+
+
+def walk_up(node, graph, prior_node=None):
+    """Create graph from a single node, searching up and down the chain
+
+    Parameters
+    ----------
+    node: Stream instance
+    graph: networkx.DiGraph instance
+    """
+    if node is None:
+        return
+    t = hash(node)
+    graph.add_node(t, stream=node)
+    if prior_node:
+        tt = hash(prior_node)
+        if graph.has_edge(t, tt):
```

Comment:
This was mostly a copy paste from the prior graphing stuff (which handled the entire graph and could have crossed itself). In the total graph stuff it was there just in case we went down and came back up the same route. I guess we shouldn't run into that situation here, although we do support cyclical graphs.